pipeline/compilers/__init__.py

```diff
@@ -116,7 +116,7 @@ def execute_command(self, command, cwd=None, stdout_captured=None):
 
         # The first element in argument_list is the program that will be executed; if it is '', then
         # a PermissionError will be raised. Thus empty arguments are filtered out from argument_list
-        argument_list = filter(None, argument_list))
```

Comment:
Something funny with the brackets. In the orginal we have one opening and two closing brackets. Now we have two opening brackets and three closing brackets.